test/e2e/nodeportlocal_test.go

```diff
@@ -0,0 +1,389 @@
+// +build !windows
+
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/nodeportlocal/k8s"
+)
+
+const (
+	defaultStartPort = 40000
+	defaultEndPort   = 41000
+	updatedStartPort = 42000
+	updatedEndPort   = 43000
+	nplSvcAnnotation = "nodeportlocal.antrea.io/enabled"
+)
+
+func getNPLAnnotation(t *testing.T, data *TestData, r *require.Assertions, testPodName string) (string, *PodIPs) {
+	var nplAnn string
+	var testPodIP *PodIPs
+	var found bool
+	var err error
+	err = wait.Poll(time.Second, defaultTimeout, func() (bool, error) {
+		updatedPod, _ := data.podWaitFor(defaultTimeout, testPodName, testNamespace, func(pod *corev1.Pod) (bool, error) {
+			return pod.Status.Phase == corev1.PodRunning, nil
+		})
+		testPodIP, err = data.podWaitForIPs(defaultTimeout, testPodName, testNamespace)
+		if err != nil {
+			t.Logf("Error when waiting for IP for Pod '%s': %v", testPodName, err)
+			return false, nil
+		}
+		r.Nil(err, "Failed to get Pod")
```

Comment:
you check the error 4 lines above already and return if it is not `nil`

Reply:
I'll remove this check.